uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm

```diff
@@ -457,11 +453,8 @@ sub createTarchiveArray {
 
     my $this = shift;
     my %tarchiveInfo;
-    my ($tarchive,$globArchiveLocation) = @_;
-    my $where = "ArchiveLocation='$tarchive'";
-    if ($globArchiveLocation) {
-        $where = "ArchiveLocation LIKE '%".basename($tarchive)."'";
-    }
+    my ($tarchive) = @_;
+    my $where = "ArchiveLocation LIKE '%" . quotemeta(basename($tarchive)) . "%'";
```

Comment:
So if `$tarchive` is `/my/path/to/tarchive` and `ArchiveLocation` is `/x/y/z/PREFIXtarchiveSUFFIX` then where condition will evaluate to true, which is wrong IMHO.

How about

```
my $where = "CONCAT('/', ArchiveLocation) LIKE '%/" . quotemeta(basename($tarchive)) ."'";
```

The `CONCAT` makes sure that ArchiveLocation *always* contains a slash. Also, we should use a prepared statement so the where will contain a placeholder instead of `quotemeta(basename($tarchive))`